### ISSUE TITLE
display a real message error on ldap connection fail

### ISSUE THREAD

**Issue by Psycojoker** [YunoHost/issues#1158]
Right now we have this thing:

    moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

And I'm pretty sure that means "I failed to conect to ldap or have lost my connection" and that a pretty shitty and horrible error message.

At least we need to investigate on how to fix this crap.

**Comment by alexAubin** [comment]
What I dont understand with this is that it should like, totally crash all following things, i.e. if the code wants to list domains, it should use the ldap authenticat things, but if it's `None` that should crash ... so I really don't understand for now what's going on x_x.

Also I asked people who encountered this to try to restart slapd (and yunohost-api for those on the webadmin) but that didnt fix the issue apparently x_x

**Comment by Psycojoker** [comment]
Yeah, I don't understand neither :/

FYI it's from this log https://paste.yunohost.org/kupefitetu but I haven't took the time to dig into it

```
    2018-06-30 09:40:36,681 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:36,703 INFO     moulinette.actionsmap process - processing action [533.29]: yunohost.tools.diagnosis

    2018-06-30 09:40:37,242 INFO     moulinette.core translate - untranslated key 'service_description_php7.0-fpm' for locale 'fr'

    2018-06-30 09:40:37,286 ERROR    moulinette.core translate - unable to retrieve key 'service_description_seafile-server' for default locale 'en'

    Traceback (most recent call last):

      File "/usr/lib/python2.7/dist-packages/moulinette/interfaces/api.py", line 268, in wrapper

        curr_v = params[k]

    KeyError: 'locale'

    2018-06-30 09:40:38,614 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:38] "GET /diagnosis?locale=fr HTTP/1.1" 200 1551 1.933552

    2018-06-30 09:40:39,981 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:39,982 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:40,002 INFO     moulinette.actionsmap process - processing action [533.30]: yunohost.tools.update

    2018-06-30 09:40:41,231 SUCCESS  yunohost.app success - [533.30] La liste d’applications yunohost a été récupérée

    2018-06-30 09:40:41,457 SUCCESS  yunohost.app success - [533.30] La liste d’applications community a été récupérée

    2018-06-30 09:40:41,597 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:41] "PUT /update HTTP/1.1" 200 12523 1.616711

    2018-06-30 09:40:42,050 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:42,050 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:42,068 INFO     moulinette.actionsmap process - processing action [533.31]: yunohost.user.list

    2018-06-30 09:40:42,071 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:42] "GET /users?locale=fr HTTP/1.1" 200 493 0.021357

    2018-06-30 09:40:42,155 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:42,156 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:42,179 INFO     moulinette.actionsmap process - processing action [533.32]: yunohost.tools.diagnosis

    2018-06-30 09:40:42,633 INFO     moulinette.core translate - untranslated key 'service_description_php7.0-fpm' for locale 'fr'

    2018-06-30 09:40:42,671 ERROR    moulinette.core translate - unable to retrieve key 'service_description_seafile-server' for default locale 'en'

    Traceback (most recent call last):

      File "/usr/lib/python2.7/dist-packages/moulinette/interfaces/api.py", line 268, in wrapper

        curr_v = params[k]

    KeyError: 'locale'

    2018-06-30 09:40:44,044 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:44] "GET /diagnosis?locale=fr HTTP/1.1" 200 1551 1.888452

    2018-06-30 09:40:44,048 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:44] "GET /installed HTTP/1.1" 200 159 0.001351

    2018-06-30 09:40:45,147 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:45,148 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:45,169 INFO     moulinette.actionsmap process - processing action [533.33]: yunohost.user.list

    2018-06-30 09:40:45,172 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:45] "GET /users?locale=fr HTTP/1.1" 200 493 0.024710

    2018-06-30 09:40:45,255 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:45,255 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:45,273 INFO     moulinette.actionsmap process - processing action [533.34]: yunohost.tools.diagnosis

    2018-06-30 09:40:45,697 INFO     moulinette.core translate - untranslated key 'service_description_php7.0-fpm' for locale 'fr'

    2018-06-30 09:40:45,735 ERROR    moulinette.core translate - unable to retrieve key 'service_description_seafile-server' for default locale 'en'

    Traceback (most recent call last):

      File "/usr/lib/python2.7/dist-packages/moulinette/interfaces/api.py", line 268, in wrapper

        curr_v = params[k]

    KeyError: 'locale'

    2018-06-30 09:40:47,052 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:47] "GET /diagnosis?locale=fr HTTP/1.1" 200 1551 1.797024

    2018-06-30 09:40:47,057 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:47,058 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:47,077 INFO     moulinette.actionsmap process - processing action [533.35]: yunohost.tools.update

    2018-06-30 09:40:48,284 SUCCESS  yunohost.app success - [533.35] La liste d’applications yunohost a été récupérée

    2018-06-30 09:40:48,513 SUCCESS  yunohost.app success - [533.35] La liste d’applications community a été récupérée

    2018-06-30 09:40:48,611 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:48] "PUT /update HTTP/1.1" 200 12523 1.557231

    2018-06-30 09:40:48,966 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:48] "GET /installed HTTP/1.1" 200 159 0.001140

    2018-06-30 09:40:58,364 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:40:58,397 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:58] "POST /login HTTP/1.1" 200 406 0.033792

    2018-06-30 09:40:58,744 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:40:58] "GET /version?locale=fr HTTP/1.1" 200 357 0.234446

    2018-06-30 09:41:03,712 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:03,732 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:41:03] "POST /login HTTP/1.1" 200 406 0.020558

    2018-06-30 09:41:04,034 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:41:04] "GET /version?locale=fr HTTP/1.1" 200 357 0.188263

    2018-06-30 09:41:05,590 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:05,591 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:05,612 INFO     moulinette.actionsmap process - processing action [533.36]: yunohost.tools.update

    2018-06-30 09:41:06,921 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:06,922 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:06,942 INFO     moulinette.actionsmap process - processing action [533.37]: yunohost.tools.update

    2018-06-30 09:41:06,920 SUCCESS  yunohost.app success - [533.36] La liste d’applications yunohost a été récupérée

    2018-06-30 09:41:08,422 SUCCESS  yunohost.app success - [533.37] La liste d’applications yunohost a été récupérée

    2018-06-30 09:41:08,653 SUCCESS  yunohost.app success - [533.37] La liste d’applications community a été récupérée

    2018-06-30 09:41:09,022 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:41:09] "PUT /update HTTP/1.1" 200 12523 3.433307

    2018-06-30 09:41:08,881 SUCCESS  yunohost.app success - [533.37] La liste d’applications community a été récupérée

    2018-06-30 09:41:09,120 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:41:09] "PUT /update HTTP/1.1" 200 12523 2.199157

    2018-06-30 09:41:09,135 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:09,135 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:09,154 INFO     moulinette.actionsmap process - processing action [533.38]: yunohost.tools.update

    2018-06-30 09:41:10,659 SUCCESS  yunohost.app success - [533.38] La liste d’applications yunohost a été récupérée

    2018-06-30 09:41:10,885 SUCCESS  yunohost.app success - [533.38] La liste d’applications community a été récupérée

    2018-06-30 09:41:10,990 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:41:10] "PUT /update HTTP/1.1" 200 12523 1.855569

    2018-06-30 09:41:10,992 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:10,992 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:11,013 INFO     moulinette.actionsmap process - processing action [533.39]: yunohost.tools.update

    2018-06-30 09:41:12,352 SUCCESS  yunohost.app success - [533.39] La liste d’applications yunohost a été récupérée

    2018-06-30 09:41:12,579 SUCCESS  yunohost.app success - [533.39] La liste d’applications community a été récupérée

    2018-06-30 09:41:12,697 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:41:12] "PUT /update HTTP/1.1" 200 12523 1.705679

    2018-06-30 09:41:12,698 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:12,699 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:12,721 INFO     moulinette.actionsmap process - processing action [533.40]: yunohost.tools.update

    2018-06-30 09:41:13,948 SUCCESS  yunohost.app success - [533.40] La liste d’applications yunohost a été récupérée

    2018-06-30 09:41:14,173 SUCCESS  yunohost.app success - [533.40] La liste d’applications community a été récupérée

    2018-06-30 09:41:14,285 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:41:14] "PUT /update HTTP/1.1" 200 12523 1.587296

    2018-06-30 09:41:16,733 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:41:16] "GET /version?locale=fr HTTP/1.1" 200 357 0.227818

    2018-06-30 09:41:16,734 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:16,735 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:16,755 INFO     moulinette.actionsmap process - processing action [533.41]: yunohost.tools.update

    2018-06-30 09:41:17,963 SUCCESS  yunohost.app success - [533.41] La liste d’applications yunohost a été récupérée

    2018-06-30 09:41:18,188 SUCCESS  yunohost.app success - [533.41] La liste d’applications community a été récupérée

    2018-06-30 09:41:18,328 INFO     geventwebsocket.handler log_request - 127.0.0.1 - - [2018-06-30 09:41:18] "PUT /update HTTP/1.1" 200 12523 1.594283

    2018-06-30 09:41:31,080 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:31,081 WARNING  moulinette.authenticator.ldap is_authenticated - Error during ldap authentication process: 'NoneType' object has no attribute 'whoami_s'

    2018-06-30 09:41:31,102 INFO     moulinette.actionsmap process - processing action [533.42]: yunohost.tools.upgrade

    2018-06-30 09:41:31,658 INFO     yunohost.tools tools_upgrade - [533.42] Mise à jour des paquets...

    2018-06-30 09:41:40,407 INFO     yunohost.tools tools_upgrade - [533.42] Terminé

    2018-06-30 09:41:40,408 SUCCESS  yunohost.tools success - [533.42] Le système a été mis à jour

**Comment by 4abhinavjain** [comment]
I am still seeing this error on 3.2.1 (Stable).

https://paste.yunohost.org/duguraketo

**Comment by alexAubin** [comment]
Fixed in 3.3.0 : https://github.com/YunoHost/moulinette/pull/173
